packages/notifier/src/subscriber.js

```diff
@@ -64,6 +64,11 @@ const makeSubscriptionIterator = tailP => {
  * @returns {SubscriptionRecord<T>}
  */
 const makeSubscriptionKit = (...optionalInitialState) => {
+  assert(
+    // optionalInitialState allows just one state and has to be an array to allow specifying 'undefined'
+    optionalInitialState.length === 0 || optionalInitialState.length === 1,
+    'Too many arguments',
+  );
```

Comment:
I appreciate the idea of not allowing extra arguments, but that's not a conventional behaviour for JS functions.  There are users of this API outside of the SDK, so I don't want to do that.

The comment is valuable, though.  Maybe add it to the JSDoc for this parameter.
```suggestion
  // optionalInitialState allows just one state and has to be an array to allow specifying 'undefined'
```

Reply:
Agree it's normal that JS functions allow extra arguments. But they also don't have JSDoc saying that there's a rest parameter of arbitrary length. I agree the comment is best placed at the param declaration. I also think given that comment it's fair to throw when there's a bad input, just like we do in so many other instances. 

to grab one as an example,
```
const assertParamChangeIssue = issue => {
  assert(issue, X`argument to assertParamChangeIssue cannot be null`);
```
this is about the same,
```
  assert(
    optionalInitialState.length === 0 || optionalInitialState.length === 1,
    'optionalInitialState must have zero or one elements',
  );
```
Granted it makes the function fail on extra arguments which most JS functions don't bother to do, but the rest parameter turns the end of the parameter list into a param which can be valid or invalid.

If that's not convincing let me know and I will take it out.